### PR TITLE
Create topic

### PR DESCRIPTION
**Pull request by HuziMuzi**
Add select and create topic for question

image

### CODE REVIEW

**Review by KhaninPasha0488 — APPROVED** [review]

File: src/screens/CreateTest/CreateQuestion/TopicQuestion/index.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+type InputFieldType = {
+  topicName: string;
+};
+
+export const TopicQuestion = ({setValue}: TopicQuestionPropsType) => {
+  const dispatch = useAppDispatch();
+  const {control, handleSubmit, reset} = useForm<InputFieldType>();
+  const [topics, setTopics] = useState<TopicType[]>([]);
+  const [isTopic, setIsTopic] = useState<string>();
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/TopicQuestion/index.tsx:25]: without `is`

File: src/screens/CreateTest/CreateQuestion/TopicQuestion/index.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    const newTopic = await dispatch(createTopic(topicName)).unwrap();
+    const allTopics = await dispatch(getTopics()).unwrap();
+    setTopics(allTopics);
+    setValue('topicId', newTopic.id);
+    setIsTopic(newTopic.title);
+    reset({topicName: ''});
+  };
+
+  const onSelectTopicPressed = (title: string) => {
+    const isTopicId = +topics.find(el => el.title === title)!.id;
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/TopicQuestion/index.tsx:37]: without `is`

File: src/screens/CreateTest/CreateQuestion/TopicQuestion/index.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      });
+  }, [dispatch]);
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.selectContainer}>
+        <AppSelect
+          data={topics.map((el: TopicType) => el.title)}
+          size="m"
+          type="primary"
```
Comment (antoniosk10) [inline src/screens/CreateTest/CreateQuestion/TopicQuestion/index.tsx:56]: enum

Comment (HuziMuzi) [reply src/screens/CreateTest/CreateQuestion/TopicQuestion/index.tsx:56]: This will be added in another PR, I will correct it here later

**Review by antoniosk10 — APPROVED** [review]

**Review by KhaninPasha0488 — DISMISSED** [review]

**Review by KhaninPasha0488 — APPROVED** [review]

**Review by antoniosk10 — APPROVED** [review]

**Pull request merged**